src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/CallableMixedTest.java

```diff
@@ -31,79 +32,71 @@
-        callableStatement.setObject((int) 4, Short.valueOf("-5372"), (int) 5);
+            try (CallableStatement callableStatement = connection.prepareCall("{  ? = CALL " + procName + " (?, ?, ?, ?) }")) {
+                callableStatement.registerOutParameter((int) 1, (int) 4);
+                callableStatement.setObject((int) 2, Integer.valueOf("31"), (int) 4);
+                callableStatement.registerOutParameter((int) 3, (int) 4);
+                callableStatement.registerOutParameter((int) 5, java.sql.Types.BINARY);
+                callableStatement.registerOutParameter((int) 5, (int) 5);
+                callableStatement.setObject((int) 4, Short.valueOf("-5372"), (int) 5);
 
-        // get results and a value
-        ResultSet rs = callableStatement.executeQuery();
-        rs.next();
+                // get results and a value
+                ResultSet rs = callableStatement.executeQuery();
+                rs.next();
 
-        assertEquals(rs.getInt(1), 0, "Received data not equal to setdata");
-        assertEquals(callableStatement.getInt((int) 5), -5372, "Received data not equal to setdata");
+                assertEquals(rs.getInt(1), 0, "Received data not equal to setdata");
+                assertEquals(callableStatement.getInt((int) 5), -5372, "Received data not equal to setdata");
 
-        // do nothing and reexecute
-        rs = callableStatement.executeQuery();
-        // get the param without getting the resultset
-        rs = callableStatement.executeQuery();
-        assertEquals(callableStatement.getInt((int) 1), -2147483648, "Received data not equal to setdata");
+                // do nothing and reexecute
+                rs = callableStatement.executeQuery();
+                // get the param without getting the resultset
+                rs = callableStatement.executeQuery();
+                assertEquals(callableStatement.getInt((int) 1), -2147483648, "Received data not equal to setdata");
 
-        rs = callableStatement.executeQuery();
-        rs.next();
+                rs = callableStatement.executeQuery();
+                rs.next();
 
-        assertEquals(rs.getInt(1), 0, "Received data not equal to setdata");
-        assertEquals(callableStatement.getInt((int) 1), -2147483648, "Received data not equal to setdata");
-        assertEquals(callableStatement.getInt((int) 5), -5372, "Received data not equal to setdata");
-        rs = callableStatement.executeQuery();
-        callableStatement.close();
-        rs.close();
-        stmt.close();
+                assertEquals(rs.getInt(1), 0, "Received data not equal to setdata");
+                assertEquals(callableStatement.getInt((int) 1), -2147483648, "Received data not equal to setdata");
+                assertEquals(callableStatement.getInt((int) 5), -5372, "Received data not equal to setdata");
+                rs = callableStatement.executeQuery();
+                rs.close();
+            }
+        }
```

Comment:
Before closing the try, terminate variation can be passed existing Connection & Statement objects instead of starting a new connection for dropping objects.